src/modules/aws.rs

```diff
@@ -148,13 +148,38 @@ fn alias_name(name: Option<String>, aliases: &HashMap<String, &str>) -> Option<S
 
 fn has_credential_process_or_sso(
     context: &Context,
-    aws_profile: &Option<Profile>,
+    aws_profile: Option<&Profile>,
     aws_config: &AwsConfigFile,
+    aws_creds: &AwsCredsFile,
 ) -> Option<bool> {
     let config = get_config(context, aws_config)?;
+    let credentials = get_creds(context, aws_creds);
+
+    let empty_section = ini::Properties::new();
+    // We use the aws_profile here because `get_profile_config()` treats None
+    // as "special" and falls back to the "[default]"; otherwise this tries
+    // to look up "[profile default]" which doesn't exist
+    let config_section = get_profile_config(config, aws_profile).or(Some(&empty_section))?;
+
+    let default_profile = "default".to_string();
+    let credential_profile = aws_profile.unwrap_or(&default_profile);
+
+    // credentials are optional; if we can't find the credential file,
+    // we work with just the config and treat the credential section as
+    // empty
+    let credential_section = match credentials {
+        Some(credentials) => credentials
+            .section(Some(credential_profile))
+            .or(Some(&empty_section))?,
+        None => &empty_section,
+    };
 
-    let section = get_profile_config(config, aws_profile)?;
-    Some(section.contains_key("credential_process") || section.contains_key("sso_start_url"))
+    Some(
+        config_section.contains_key("credential_process")
+            || config_section.contains_key("sso_start_url")
+            || credential_section.contains_key("credential_process")
+            || credential_section.contains_key("sso_start_url"),
```

Comment:
`has_credential_process_or_sso` has been mostly concerned with the config file and `has_defined_credentials` with the credentials file. Given that, please move the `has_defined_credentials` into `has_defined_credentials` or change this function to first do the check on the config file and either return on success, or follow up with the credentials file check.

Reply:
That doesn't quite feel right to me. That checks for an actual active credential, which in this case would require me to _invoke_ the credential process under the hood... _not_ something we want to do in the prompt. This is just looking for the configuration that _enables_ credentials, in the second place where it can be found. The AWS CLI allows for it to be configured in two places (which is a bad idea, for sure!) but right now Starship only looks in one of them.

If I'm going to change that, I think the name of those methods should change as well. I'll update the PR with the changes from the below, but I think this method's the right place for the code. If you insist otherwise, I'll move it, but it feels wrong to me.

Reply:
I would still prefer If you changed this code to do the checks sequentially, to only read each file based on necessity. But since it looks like they both will be read in most cases either way, keeping it this way would be fine too.

Reply:
That was my thinking too; the most common configuration will have both files read, so I don't think there's a lot of benefit or optimization in making that more complex. Doubly so because starship caches the read, so it's still pretty safe.